src/gui/sdl_mapper.cpp

```diff
@@ -921,6 +925,49 @@ class CStickBindGroup : public CBindGroup {
 			return "[missing joystick]";
 	}
 
+	void SetTriggerButtonFor(const char* button_name, const char* sdl_mapping)
+	{
+		// Part of the string we care about is in the format of:
+		// "button_name" + ':' + single character button type (a for axis, b for button, h for hat) + axis number
+
+		const char* substring = strstr(sdl_mapping, button_name);
+		if (!substring) {
+			return;
+		}
+		substring += strlen(button_name);
+		if (*substring != 'a') {
+			// Not an axis so no need to do anything
+			return;
+		}
+		++substring;
+		if (!(*substring >= '0' && *substring <= '9')) {
+			// Safety check, this means the string is an invalid format
+			return;
+		}
+		int axis_number = 0;
+		while (*substring >= '0' && *substring <= '9') {
+			axis_number *= 10;
+			axis_number += *substring - '0';
+			++substring;
+		}
+		if (axis_number >= 0 && axis_number < MAXAXIS) {
+			is_trigger[axis_number] = true;
+		}
+	}
+
+	void DetectTriggerButtons()
+	{
+		char* sdl_mapping = SDL_GameControllerMappingForDeviceIndex(stick_index);
+		if (!sdl_mapping) {
+			return;
+		}
+		SetTriggerButtonFor("lefttrigger:", sdl_mapping);
+		SetTriggerButtonFor("righttrigger:", sdl_mapping);
+		SDL_free(sdl_mapping);
+	}
+
+	bool is_trigger[MAXAXIS] = {};
```

Comment:
As long as I've used DOSBox, I've never been able to get consistent mappings across controller brands: if you change controller brands then you have to re-do your mappings from the ground up.

I understand this is why eXoDOS can't ship joystick mappings.

If your approach here can also cover the buttons and axis, then we'd have 100% normalization, and finally users would have portable and sharable mappings independent of the controller make and model.

(I realize there's a plan to re-write the controller, but looking at this code - it's something that could be used in 0.83 and used in the re-write, too)

Reply:
This doesn't actually address that problem.  It only marks what axis are trigger buttons because they need a special case to handle their input.

To get real controller normalization we would need to use SDL's gamepad API everywhere (this is what provides the mappings that I use here) which would mean both a refactor of the mapper code and breaking everyone's existing mapper files.

Reply:
Thanks for that explanation; (darn; was hoping there was an easy silver bullet in here! I guess the move to the game controller DB is still the one true solution)

Reply:
Let's not overdo it @interloper98. This will all be nuked. We only do surgical fixes until this is rebuilt, and this PR already goes beyond that (but it's a welcome change).

In the new mapper, we'll have proper SDL ControllerDB support. Let's not invent alternative home-grown mechanisms until then, that's my view.

Reply:
Yeah; I was seeing a solution when there wasn't one. 

1. Load DB
2. Assign axis from DB
3. Assign buttons from DB
4. Assign triggers from DB

(In about as many lines of code, too). But turns out it's not as simple or possible; Here's to dreaming! 🤷‍♂️ 😆